### PR TITLE
Add Build & Publish Docker Images Workflow

### PR DESCRIPTION
**Pull request by ellioty**
- Add reusable Build & Publish Docker images workflow

### CODE REVIEW

File: .github/workflows/publish-docker-image.yml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        with:
+          registry: ${{ env.GITHUB_CONTAINNER_REGISTRY_URL }}
+          username: ${{ secrets.GITHUB_CONTAINER_USERNAME }}
+          password: ${{ secrets.GITHUB_CONTAINER_PASSWORD }}
+
+      - name: Build and publish image to Github Container Registry
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          file: deployment/Dockerfile
```
Comment (Voxelot) [inline .github/workflows/publish-docker-image.yml:73]: We should make this a configurable input, as some projects could have multiple dockerfiles to publish etc

Comment (ellioty) [reply .github/workflows/publish-docker-image.yml]: Just added an input with a default value of `deployment/Dockerfile`

File: .github/workflows/publish-docker-image.yml [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+on:
+  workflow_call:
+    inputs:
+      flavor:
+        description: "Defines global behaviors for tags (e.g. 'latest=auto')"
+        required: false
+        type: string
+
+      deployment_file:
```
Comment (Voxelot) [inline .github/workflows/publish-docker-image.yml:11]: What is a "deployment file"? Shouldn't this just be `dockerfile`?

Comment (ellioty) [reply .github/workflows/publish-docker-image.yml]: Updated the variable name

**Review by Voxelot — APPROVED** [review]

**Pull request merged**